### ISSUE TITLE
Eliminate render-blocking resources

### ISSUE THREAD

**Issue by unclesam2024** [web-infrastructure/doc_architecture#34]
Avoid using resources which 'block', i.e. which stop other resources from loading at the same time.

This check looks for two potential issues:

A `<script>` tag which:

Is in the `<head>` of the document
Does not have a `defer` attribute
Does not have an `async` attribute
A `<link rel="stylesheet">` tag which:

- Does not have a `disabled` attribute. When this attribute is present, the browser does not download the stylesheet.

- Does not have a `media` attribute that matches the user's device specifically. `media="all"` is considered render-blocking.

Solution
In both cases, consider inlining any critical CSS or JavaScript resources, and
deferring all non-critical resources.

[Learn more about blocking resources](https://web.dev//render-blocking-resources//).
